py3status/modules/weather_owm.py

```diff
@@ -0,0 +1,750 @@
+    format_rain:
+        {amount} Rainfall in the specified unit
+        {unit} The unit specified
+    format_snow:
+        {amount} Snowfall in the specified unit
+        {unit} The unit specified
+    format_temperature:
+        {current} Current temperature
+        {max} Maximum temperature in the configured unit
+        {min} Minimum temperature
+        {unit} The unit specified
+    format_wind:
+        {degree} Current wind heading
+        {gust} Wind gusts speed in the specified unit
+        {speed} Wind speed
+        {unit} The unit specified
+    format only:
+        {city} The name of the city where the weather is
+        {forecast} Output of format_forecast
+    format, format_forecast:
+        {clouds} Output of format_clouds
+        {description} Natural description of the current weather
+        {humidity} Output of format_humidity
+        {main} Short description of the current weather
+        {pressure} Output of format_pressure
+        {snow} Output of format_snow
+        {sunrise} Output of format_sunrise
+        {sunset} Output of format_sunset
+        {temperature} Output of format_temperature
+        {wind} Output of format_wind
+
+Example configuration:
+```
+weather_owm {
+  api_key = '...'
+
+  icons {
+    i200 = "☔"
+    i230_i232 = "🌧"
+
+    clouds = "☁"
+  }
+}
+```
+Outputs: 🌫 ○ 59° foggy ⛅ ☼ 🌧`
+- Currently foggy, 59° F outside, with forecast of cloudy tomorrow, sunny the
+  next day, then rainy
+
+
+@author alexoneill
+@licence MIT
+'''
+
+import datetime
+
+
+# API information
+OWM_API = '2.5'
+OWM_CURR_ENDPOINT = 'http://api.openweathermap.org/data/%s/weather?' \
+    'APPID=%s&lat=%f&lon=%f&lang=%s'
+OWM_FUTURE_ENDPOINT = 'http://api.openweathermap.org/data/%s/forecast?' \
+    'APPID=%s&lat=%f&lon=%f&lang=%s&cnt=%%d'
+IP_ENDPOINT = 'http://ip-api.com/json'
+
+# Paths of information to extract from JSON
+IP_LAT = '//lat'
+IP_LNG = '//lon'
+OWM_CLOUD_COVER = '//clouds/all'
+OWM_DESC = '//weather:0/main'
+OWM_DESC_LONG = '//weather:0/description'
+OWM_HUMIDITY = '//main/humidity'
+OWM_PRESSURE = '//main'
+OWM_RAIN = '//rain/3h'
+OWM_SNOW = '//snow/3h'
+OWM_SUNRISE = '//sys/sunrise'
+OWM_SUNSET = '//sys/sunset'
+OWM_TEMP = '//main'
+OWM_WEATHER_ICON = '//weather:0/id'
+OWM_WIND = '//wind'
+
+# Units constants
+RAIN_UNITS = set(['cm', 'ft', 'in', 'mm', 'm', 'yd'])
+SNOW_UNITS = RAIN_UNITS
+TEMP_UNITS = set(['c', 'f', 'k'])
+WIND_UNITS = set(['fsec', 'msec', 'mph', 'kmh'])
+
+# Conversion factors
+FT_FROM_METER = 3.28084
+IN_FROM_MM = 0.0393701
+KMH_FROM_MSEC = 0.277778
+MPH_FROM_MSEC = 2.23694
+
+# Thresholds defaults
+THRESHOLDS = [
+    (-100, '#0FF'),
+    (0, '#00F'),
+    (50, '#0F0'),
+    (150, '#FF0')
+]
+
+
+class OWMException(Exception):
+    pass
+
+
+class Py3status:
+
+    api_key = None
+    cache_timeout = 600
+    forecast_days = 3
+    forecast_include_today = False
+    forecast_text_separator = ' '
+    format = '{city}: {icon} {temperature} {main} {forecast}'
```

Comment:
Hmm, not sure. I'm fine removing `{main}`, especially since it seems to mimic the longer form. Even then, `{description}` is never really that verbose, so maybe it makes sense...